cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,217 @@
+"""
+Views for the maintenance app.
+"""
+import logging
+from django.db import transaction
+from django.core.urlresolvers import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import View
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from contentstore.management.commands.utils import get_course_versions
+from util.views import require_global_staff
+
+
+log = logging.getLogger(__name__)
+
+MAINTENANCE_COMMANDS = {
+    "force_publish_course": {
+        "url": reverse_lazy("maintenance:force_publish_course"),
+        "name": _("Force Publish Course"),
+        "slug": "force_publish_course",
+        "description": _("Force publish course."),
+    },
+}
+
+
+COURSE_KEY_ERROR_MESSAGES = {
+    'empty_course_key': _('Please provide course id.'),
+    'invalid_course_key': _('Invalid course key.'),
+    'course_key_not_found': _('No matching course found.')
+}
+
+
+def get_maintenace_urls():
+    """
+    Returns all URLs for maintenance app.
+    """
+    url_list = []
+    for key, val in MAINTENANCE_COMMANDS.items():  # pylint: disable=unused-variable
+        url_list.append(val['url'])
+    return url_list
+
+
+class MaintenanceIndexView(View):
+    """
+    View for maintenance dashboard, used by the escalation team.
+    """
+
+    @method_decorator(require_global_staff)
+    def get(self, request):
+        """Render the maintenance index view. """
+        return render_to_response('maintenance/index.html', {
+            "commands": MAINTENANCE_COMMANDS,
+        })
+
+
+class MaintenanceBaseView(View):
+    """
+    Base class for Maintenance views.
+    """
+
+    template = 'maintenance/container.html'
+
+    def __init__(self, command=None):
+        self.context = {
+            'command': command if command else '',
+            'form_data': {},
+            'error': False,
+            'msg': ''
+        }
+
+    def render_response(self):
+        """ Renders response."""
+        return render_to_response(self.template, self.context)
+
+    @method_decorator(require_global_staff)
+    def get(self, request):
+        """Render get view."""
+        return self.render_response()
+
+    def validate_course_key(self, course_key, branch=None):
+        """Validates course_key and updates context, returns usage_key, that would be used by maintenance app views."""
+        course_usage_key = None
+        if not course_key:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['empty_course_key']
+            return course_usage_key
+
+        try:
+            if branch == "published":
+                course_key += "+branch@published-branch"
```

Comment:
instead of manipulating the string, use methods already defined on the course key:
```python
In [9]: ck = CourseKey.from_string("course-v1:a+b+c")

In [10]: ck.for_branch?
Signature: ck.for_branch(branch)
Docstring: Return a new CourseLocator for another branch of the same course (also version agnostic)
File:      ~/venvs/edxapp/local/lib/python2.7/site-packages/opaque_keys/edx/locator.py
Type:      instancemethod

In [11]: unicode(ck.for_branch('published-branch'))
Out[11]: u'course-v1:a+b+c+branch@published-branch'
```